src/lib/videotagging_extensions/detection.js

```diff
@@ -104,14 +88,38 @@ function Detection(videotagging, visitedFrames) {
                 if (err) {
                     cb(err);
                 }
+                //resolve export until
+                var isLastFrame;
+                if (exportUntil === "tagged") {
```

Comment:
I'd recommend you break these if blocks into smaller functions that can be called.

Reply:
As in make it so that the isLastFrame functions can be called at assignment instead of anonymously assigned there?